frontend/src/scenes/dashboard/DashboardTemplateEditorLogic.ts

```diff
@@ -0,0 +1,116 @@
+import { editor } from 'monaco-editor'
+import { lemonToast } from '@posthog/lemon-ui'
+import { actions, connect, kea, listeners, path, reducers } from 'kea'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { DashboardTemplateType } from '~/types'
+import { dashboardTemplatesLogic } from './dashboards/templates/dashboardTemplatesLogic'
+
+import type { dashboardTemplateEditorLogicType } from './DashboardTemplateEditorLogicType'
+
+export const dashboardTemplateEditorLogic = kea<dashboardTemplateEditorLogicType>([
+    path(['scenes', 'dashboard', 'NewDashboardTemplateLogic']),
+    connect(dashboardTemplatesLogic),
+    actions({
+        setDashboardTemplateJSON: (dashboardTemplateJSON: string) => ({ dashboardTemplateJSON }),
+        setOpenNewDashboardTemplateModal: (openNewDashboardTemplateModal: boolean) => ({
+            openNewDashboardTemplateModal,
+        }),
+        createDashboardTemplate: (dashboardTemplateJSON: string) => ({ dashboardTemplateJSON }),
+        setDashboardTemplateId: (id: string | undefined) => ({ id }),
+        closeNewDashboardTemplateModal: true,
+        updateValidationErrors: (markers: editor.IMarker[] | undefined) => ({ markers }),
+    }),
+    reducers({
+        dashboardTemplateJSON: [
+            '' as string,
+            {
+                setDashboardTemplateJSON: (_, { dashboardTemplateJSON }) => dashboardTemplateJSON,
+            },
+        ],
+        validationError: [
+            '' as string,
```

Comment:
We might be better with `validationErrors` as a `string[]` then the UI can lay them out nicely or count them. 

The `disabledReason` on buttons can be something like `There are ${validationErrors.length} errors to resolve`